packages/web/src/components/track/PremiumConditionsPill.tsx

```diff
@@ -39,12 +61,24 @@ export const PremiumConditionsPill = ({
       : messages.locked
   }
 
-  const colorStyle = isPurchase ? styles.premiumContent : styles.gatedContent
+  const handleClick = useCallback(
```

Comment:
not sure i love that I'm bifurcating the style of component - it used to be presentational down from the `ConnectedTrackTile`. I can always bubble this up and plumb it through if that's the preferred pattern

Reply:
Yeah might be nice since this component is meant to render a particularly-styled button if it just accepted an `onClick` and the parent component handled it by launching the modal

Reply:
done